tasks/ceph_deploy.py

```diff
@@ -174,12 +195,12 @@ def build_ceph_cluster(ctx, config):
                 ceph_branch = '--{var}={val}'.format(var=var, val=val)
         node_dev_list = []
         all_nodes = get_all_nodes(ctx, config)
-        mds_nodes = get_nodes_using_roles(ctx, config, 'mds')
+        mds_nodes = get_nodes_using_role(ctx, 'mds')
         mds_nodes = " ".join(mds_nodes)
-        mon_node = get_nodes_using_roles(ctx, config, 'mon')
+        mon_node = get_nodes_using_role(ctx, 'mon')
         mon_nodes = " ".join(mon_node)
         new_mon = './ceph-deploy new'+" "+mon_nodes
-        install_nodes = './ceph-deploy install '+ceph_branch+" "+all_nodes
+        install_nodes = './ceph-deploy install ' + (ceph_branch if ceph_branch else "--dev=master") + " " + all_nodes
```

Comment:
I only know a little bit about this test suite, but I want to confirm this change.  Right now, if no branch was given it would default to whatever the most recent stable release is (giant, hammer, etc.).  I doubt most tests really rely on this, as you should be explicit what version you want.  However, this will change the default behavior to install master.

I see no problem with that, but it seems like an unrelated change.

Additionally, would this behavior be better captured ten lines up where ceph_branch is initialized and set?  Seems clearer if this was done as an "else" to the "if config.get('branch') is not None"...